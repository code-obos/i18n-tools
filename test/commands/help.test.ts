import { afterEach, describe, expect, it, vi } from 'vitest';
import Help from '../../src/commands/help';

describe('help command', () => {
    afterEach(() => {
        vi.restoreAllMocks();
    });
    it('should print help message', async () => {
        const spy = vi.spyOn(console, 'log');
        await Help.run();

        expect(spy).toHaveBeenCalledOnce();
        expect(spy.mock.lastCall).toMatchSnapshot();
    });
});
