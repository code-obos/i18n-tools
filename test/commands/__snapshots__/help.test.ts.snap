// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`help command > should print help message 1`] = `
[
  "Utility for building and compiling I18N bundles

i18n-tool build <srcDir> <outDir> [...options]
i18n-tool watch <srcDir> <outDir> [...options]
i18n-tool validate <srcDir>
i18n-tool fix <srcDir> <outDir>
i18n-tool help

Options:
    -f, --format:   Output format; script, json, formatjs
                    Default: formatjs
    
    --typescript:   Output (if not json format) should be typescript
                    Default: false
    
    --strict        Run validation when building bundles.
                    Default: false
    
    --ast           Generated compiled bundles
                    Default: false
                    
    --lut           Generated look-up-table
                    Default: false",
]
`;
